holoplot/converter.py

```diff
@@ -0,0 +1,652 @@
+except ImportError:
+    xr = None
+
+try:
+    import streamz.dataframe as sdf
+except ImportError:
+    sdf = None
+
+try:
+    from intake.source.base import DataSource
+except ImportError:
+    DataSource = None
+
+try:
+    import dask.dataframe as dd
+except ImportError:
+    dd = None
+
+try:
+    import bokeh
+    if LooseVersion(bokeh.__version__) <= '0.12.14':
+        import warnings
+        # Ignore NumPy future warnings triggered by bokeh
+        warnings.simplefilter(action='ignore', category=FutureWarning)
+except:
+    pass
+
+renderer = hv.renderer('bokeh')
+
+
+class StreamingCallable(Callable):
+    """
+    StreamingCallable is a DynamicMap callback wrapper which keeps
+    a handle to start and stop a dynamic stream.
+    """
+
+    periodic = param.Parameter()
+
+    def clone(self, callable=None, **overrides):
+        """
+        Allows making a copy of the Callable optionally overriding
+        the callable and other parameters.
+        """
+        old = {k: v for k, v in self.get_param_values()
+               if k not in ['callable', 'name']}
+        params = dict(old, **overrides)
+        callable = self.callable if callable is None else callable
+        return self.__class__(callable, **params)
+
+    def start(self):
+        """
+        Start the periodic callback
+        """
+        if not self.periodic._running:
+            self.periodic.start()
+        else:
+            raise Exception('PeriodicCallback already running.')
+
+    def stop(self):
+        """
+        Stop the periodic callback
+        """
+        if self.periodic._running:
+            self.periodic.stop()
+        else:
+            raise Exception('PeriodicCallback not running.')
+
+
+
+def is_series(data):
+    return (isinstance(data, pd.Series) or
+            (sdf and isinstance(data, (sdf.Series, sdf.Seriess))) or
+            (dd and isinstance(data, dd.Series)))
+
+
+def is_streamz(data):
+    return sdf and isinstance(data, (sdf.DataFrame, sdf.Series, sdf.DataFrames, sdf.Seriess))
+
+
+class HoloViewsConverter(object):
+
+    _gridded_types = ['image', 'contour', 'contourf', 'quadmesh']
+
+    def __init__(self, data, x, y, kind=None, by=None, width=700,
+                 height=300, shared_axes=False, columns=None,
+                 grid=False, legend=True, rot=None, title=None,
+                 xlim=None, ylim=None, xticks=None, yticks=None,
+                 fontsize=None, colormap=None, stacked=False,
+                 logx=False, logy=False, loglog=False, hover=True,
+                 style_opts={}, plot_opts={}, use_index=True,
+                 value_label='value', group_label='Group',
+                 colorbar=False, streaming=False, backlog=1000,
+                 timeout=1000, persist=False, use_dask=False,
+                 datashade=False, subplots=False, label=None,
+                 groupby=None, dynamic=True, index=None, show=False,
+                 crs=None, fields={}, **kwds):
```

Comment:
It would be nice if this list of supported options could be shared with the main `__init__` file; otherwise seems like it will be unlikely to stay matching over time.